delivery_carrier_service_level/models/carrier_service_level.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2020 Therp BV.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import fields, models
+
+
+class ServiceLevel(models.Model):
+    _name = "carrier.service.level"
+    _description = "carrier service levels"
+    _order = "carrier_id, code"
+
+    code = fields.Char(string="Carrier Service Level Code")
+    name = fields.Char(string="Carrier Service Level Name", required=True)
+    carrier_id = fields.Many2one("delivery.carrier", string="Carrier")
```

Comment:
Nitpicking: as this is the main field in the ordering, why not put it first in the list of fields?